test-data/unit/check-flags.test

```diff
@@ -451,6 +451,30 @@ strict_optional = False
 [[mypy-optional]
 strict_optional = True
 
+[case testPerFileStrictOptionalListItemImportOptional]
+# flags: --config-file tmp/mypy.ini
+import standard, optional
+
+[file standard.py]
+import optional
+from typing import List
+def f(x: List[int]) -> None: pass
+f(optional.x)  # OK -- in non-strict Optional context
+f(optional.y)  # OK -- in non-strict Optional context
+
+[file optional.py]
+from typing import Optional, List
+def f(x: List[int]) -> None: pass
```

Comment:
I think I left it for parallel structure with the other tests, but it doesn't actually participate in this one.